src/field/index.js

```diff
@@ -61,18 +68,38 @@ const DATE_FORMATS = [moment.ISO_8601]
 function computeFieldsFromProperties (props, template, options = {}) {
   const fields = Object.keys(props).map((key, i) => {
     const type = fieldType(props[key])
-    const field = { name: key, type: type, alias: key }
+    const field = {
+      name: key,
+      type,
+      alias: key,
+      defaultValue: null,
+      domain: null,
+      editable: false,
+      nullable: false,
+      sqlType: 'sqlTypeOther'
+    }
+
+    // Add length field to strings and dates
+    field.length = (type === 'esriFieldTypeString') ? 128 : (type === 'esriFieldTypeDate') ? 36 : undefined
 
-    if (type === 'esriFieldTypeString') field.length = 128
-    else if (type === 'esriFieldTypeDate') field.length = 36
     return field
   })
+
+  // Add OBJECTID field if not yet there
   if (template === 'layer' && Object.keys(props).indexOf('OBJECTID') < 0) {
```

Comment:
This isn't part of your PR but does this do the same thing as `props.hasOwnProperty('OBJECTID')`?

Reply:
Looks that way to me...